### PR TITLE
Random function

### PR DESCRIPTION
**Pull request by chriseppstein**
Lots of people at SassConf want this. so here it is.

### CODE REVIEW

**Comment by chriseppstein** [comment]
@nex3 Please review. You promised this would get into 3.3. Though you may have been too drunk to remember. But I have a witness: @snugug ;)

**Comment by Snugug** [comment]
I am a witness!

File: lib/sass/script/functions.rb [hunk 1]
```diff
@@ -401,6 +401,25 @@ def self.signature(method_name, arg_arity, kwarg_arity)
...
+    # Sets the random seed used by Sass's internal random number generator.
+    #
+    # This can be used to ensure consistent random number sequences which
+    # allows for consistent results when testing, etc.
+    #
+    # @param seed [Integer]
+    # @return [Integer] The same seed.
+    def self.random_seed=(seed)
+      @random_number_generator = Random.new(seed)
+      seed
```
Comment (nex3) [inline lib/sass/script/functions.rb:413]: This is unnecessary; Ruby automatically returns the right-hand side of an assignment, even if the assignment method itself doesn't return its argument. Similarly, the doc comment describing this return value is unnecessary.

File: lib/sass/script/functions.rb [hunk 2]
```diff
@@ -2168,6 +2186,25 @@ def inspect(value)
     end
     declare :inspect, [:value]
 
+    # @overload random()
+    #   Return a decimal between 0 and 1.
+    #   @param $limit The maximum of the random integer to be returned.
```
Comment (nex3) [inline lib/sass/script/functions.rb:2191]: Copy/paste error?

File: lib/sass/script/functions.rb [hunk 3]
```diff
@@ -2205,5 +2242,25 @@ def to_h(obj)
...
+      class Random
+        def initialize(seed = nil)
+          srand(seed) if seed
+        end
+
+        def rand(*args)
+          Kernel.rand(*args)
+        end
+      end
+    end
```
Comment (nex3) [inline lib/sass/script/functions.rb:2264]: Let's move this shim into `Sass::Util`.

File: lib/sass/script/functions.rb [hunk 4]
```diff
@@ -2168,6 +2186,25 @@ def inspect(value)
     end
     declare :inspect, [:value]
 
+    # @overload random()
+    #   Return a decimal between 0 and 1.
```
Comment (nex3) [inline lib/sass/script/functions.rb:2190]: Inclusive or exclusive?

File: test/sass/functions_test.rb [hunk 5]
```diff
@@ -1574,6 +1574,20 @@ def test_inspect
...
+    assert_equal "13", evaluate("random(100)")
+  end
+
+  def test_random_works_without_a_seed
+    if Sass::Script::Functions.instance_variable_defined?("@random_number_generator")
+      Sass::Script::Functions.send(:remove_instance_variable, "@random_number_generator")
+    end
+    assert_nothing_raised do
+      evaluate("random()")
+    end
```
Comment (nex3) [inline test/sass/functions_test.rb:1589]: Let's at least assert that this is within the expected range.

File: lib/sass/script/functions.rb [hunk 6]
```diff
@@ -2168,6 +2186,25 @@ def inspect(value)
...
+    #   @param $limit The maximum of the random integer to be returned.
+    #   @return [Sass::Script::Number] A decimal value.
+    # @overload random($limit)
+    #   Return an integer between 1 and $limit inclusive.
+    #   @param $limit The maximum of the random integer to be returned.
+    #   @return [Sass::Script::Number] An integer.
+    def random(limit = nil)
+      if limit
+        assert_integer limit, "limit"
```
Comment (nex3) [inline lib/sass/script/functions.rb:2199]: We should check that `limit` is greater than zero, and test that check below.

**Comment by nex3** [comment]
Done reviewing (and I remember agreeing to this, don't worry).

**Pull request merged**